app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java

```diff
@@ -6,56 +6,154 @@
     }
 
+    // TODO instead of location tuples, allow user to view start-to-end path on a map
+    // from: http://stackoverflow.com/questions/14292398/how-to-pass-data-from-2nd-activity-to-1st-activity-when-pressed-back-android
+    // author: ρяσѕρєя K
+    // retrieved on: November 7th, 2016
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        super.onActivityResult(requestCode, resultCode, intent);
+        if(requestCode == DEFAULT_REQ_CODE) {
+            if(resultCode == RESULT_OK){
+                // from LonelyTwitter
+                start = new Gson().fromJson(intent.getStringExtra("startLocation"), Location.class);
+                end = new Gson().fromJson(intent.getStringExtra("endLocation"), Location.class);
+                TextView tv = (TextView) findViewById(R.id.textView_start);
+                tv.setText("Start: (" + String.valueOf(start.getLatitude()) + ", " + String.valueOf(start.getLongitude()) + ")");
+                tv = (TextView) findViewById(R.id.textView_end);
+                tv.setText("End: (" + String.valueOf(end.getLatitude()) + ", " + String.valueOf(end.getLongitude()) + ")");
+            }
+        }
+    }
+
+    /**
+     * Set the buttons for incrementing and decrementing the fare estimate by the user holding it down
+     */
+    private void setButtons() {
+        ImageButton fareUpButton = (ImageButton) findViewById(R.id.imageButton_fareUp);
+        fareUpButton.setOnLongClickListener(new View.OnLongClickListener() {
+            @Override
+            public boolean onLongClick(View view) {
+                if(fareEstimated != -1) {
+                    autoIncrement = true;
+                    repeatUpdateHandler.post(new RepeatUpdater());
+                }
+                return false;
+            }
+        });
+        fareUpButton.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                if((motionEvent.getAction() == MotionEvent.ACTION_UP
+                        || motionEvent.getAction() == MotionEvent.ACTION_CANCEL) && autoIncrement) {
+                    autoIncrement = false;
+                }
+                return false;
+            }
+        });
+
+        ImageButton fareDownButton = (ImageButton) findViewById(R.id.imageButton_fareDown);
+        fareDownButton.setOnLongClickListener(new View.OnLongClickListener() {
+            @Override
+            public boolean onLongClick(View view) {
+                autoDecrement = true;
+                repeatUpdateHandler.post(new RepeatUpdater());
+                return false;
+            }
+        });
+        fareDownButton.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                if((motionEvent.getAction() == MotionEvent.ACTION_UP
```

Comment:
Can this if statement have a couple of comments? not really sure what is happening?